var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -208,7 +208,21 @@ def setup_run_environment(self, env):
         if 'process_managers=slurm' in self.spec:
             env.set('SLURM_MPI_TYPE', 'pmi2')
 
+        # Because MPI functions as a compiler, we need to treat it as one and
+        # add its compiler paths to the run environment.
+        self.setup_compiler_environment(env)
```

Comment:
I generally liked the idea of defining `setup_dependent_build_environment` in terms of calling `setup_run_environment` because I assumed any environment state you'd want to apply at runtime you'd also want to apply when building a dependent. Do you happen to know if the distinct logic here shouldn't be run in `setup_dependent_build_environment` - I'm referring specifically to:

```
        if 'process_managers=slurm' in self.spec:
             env.set('SLURM_MPI_TYPE', 'pmi2')
```

Reply:
I doubt it would cause a problem, but in an entirely theoretical sense, if you were building in a slurm allocation you might not want to change slurm variables to point to a different build of slurm  (the one mvapich depends on) instead of the one you're using. But that seems really unlikely